broker/broker-security-impl/src/test/java/org/codice/ddf/broker/security/impl/SecurityServerPluginImplTest.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.broker.security.impl;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import ddf.security.Subject;
+import ddf.security.service.SecurityManager;
+import ddf.security.service.SecurityServiceException;
+import org.apache.activemq.artemis.api.core.Message;
+import org.apache.activemq.artemis.core.message.impl.CoreMessage;
+import org.apache.activemq.artemis.core.server.ServerSession;
+import org.apache.activemq.artemis.protocol.amqp.broker.AMQPMessage;
+import org.junit.Before;
+import org.junit.Test;
+
+public class SecurityServerPluginImplTest {
+  Subject mockSubject;
+
+  SecurityManager mockSecurityManager;
+
+  ServerSession mockServerSession;
+
+  SecurityServerPluginImpl securityServerPlugin;
+
+  @Before
+  public void setVariables() throws SecurityServiceException {
+    mockSubject = mock(Subject.class);
+    mockSecurityManager = mock(SecurityManager.class);
+    when(mockSecurityManager.getSubject(any(Object.class))).thenReturn(mockSubject);
+    mockServerSession = mock(ServerSession.class);
+    when(mockServerSession.getUsername()).thenReturn("hello");
+    when(mockServerSession.getPassword()).thenReturn("world");
+
+    securityServerPlugin = new SecurityServerPluginImplTester();
+    securityServerPlugin.setSecurityManager(mockSecurityManager);
+  }
+
+  @Test
+  public void testGoodSubjectNonAmqp() throws SecurityServiceException {
+    Message message = new CoreMessage();
+
+    securityServerPlugin.beforeSend(mockServerSession, null, message, false, false);
+    assertThat(message.getStringProperty("subject"), is("Hello World"));
+  }
+
+  @Test
+  public void testGoodSubjectAmqp() {
+    Message message = new AMQPMessage(1);
+    securityServerPlugin.beforeSend(mockServerSession, null, message, false, false);
+    assertThat(message.getStringProperty("subject"), is("Hello World"));
+  }
+
+  @Test
+  public void testBadSubject() throws SecurityServiceException {
```

Comment:
Even though this is a negative test, I still think it would be nice to assert that the message does indeed not contain a value for the subject after calling `beforeSend`.

Reply:
fixed